File: david/sessions/2018/easy/35_search_insert_position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        l = len(nums)

        if target > max(nums):
            return l
        if target < min(nums):
            return 0
        for i in xrange(l):
            if nums[i] == target:
                return i
            if i + 1 < l and nums[i+1] > target:
                return i + 1
        

File: pia/2020/easy/35_search_insert_position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
  if (target <= nums[0]) {
    return 0;
  }
  if (target > nums[nums.length - 1]) {
    return nums.length;
  }

  for (let i = 1; i < nums.length; i++) {
    const curr = nums[i];

    if (curr === target) {
      return i;
    }

    const left = nums[i - 1];
    if (target > left && target < curr) {
      return i;
    }
  }
};
